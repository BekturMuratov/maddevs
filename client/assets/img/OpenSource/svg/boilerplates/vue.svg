<svg xmlns="http://www.w3.org/2000/svg" width="336" height="125" fill="none" xmlns:v="https://vecta.io/nano"><g clip-path="url(#A)"><path d="M206.784.001l-38.797 67.203L129.191.001H-.008l167.996 290.997L335.984.001h-129.2z" fill="#41b883"/><path d="M206.785.001l-38.797 67.203L129.191.001h-62l100.798 174.595L268.786.001h-62.001z" fill="#34495e"/></g><defs><clipPath id="A"><path fill="#fff" d="M0 0h336v291H0z"/></clipPath></defs></svg>