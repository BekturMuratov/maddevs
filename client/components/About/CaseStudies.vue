<template>
  <section
    id="case-studies"
    class="case-studies"
    :class="{ 'case-studies--projects': currentPageName === 'projects' }"
  >
    <div class="container">
      <div>
        <h2
          class="case-studies__main-title"
          :class="`case-studies__main-title-${currentPageName}`"
        >
          Case <br v-if="currentPageName === 'projects'"> Studies
        </h2>
        <div class="case-studies__fire-icon" v-if="currentPageName === 'projects'"></div>
      </div>
      <div class="projects">
        <SingleProject
          v-for="(project, i) in projects"
          :key="i"
          :project="project"
        />
      </div>
    </div>
  </section>
</template>

<script>
import SingleProject from '@/components/About/SingleProject';

export default {
  name: 'CaseStudies',
  components: {
    SingleProject
  },
  mounted() {
    if ($nuxt.$route.name) {
      this.currentPageName = $nuxt.$route.name;
    }
  },
  data() {
    return {
      currentPageName: '',
      projects: [
        {
          projectName: 'nambafood',
          logoImg: 'nambafoodColored',
          link: '/case-studies/namba-food/',
          targetBlank: false,
          contributionWidgetColors: ['#ec1c24', '#96969c'],
          projectBackground: 'nambafoodBackground',
          fileExtension: 'png',
          projectColor: '#f8f7f6',
          projectTitle: 'Contactless Delivery Service',
          projectDescription: 'Mad Devs created the Namba Food delivery service from scratch. The solution orchestrates feature-rich apps for couriers, end-users, and business owners.',
          alt: 'Namba Food Contactless Delivery Service`s Logo.'
        },
        {
          projectName: 'teacherly',
          logoImg: 'sjmcIcon',
          link: '/case-studies/sir-john-monash-centre/',
          targetBlank: false,
          contributionWidgetColors: ['#96969c', '#96969c'],
          projectBackground: 'sjmcBackground',
          fileExtension: 'jpg',
          projectColor: '#221F20',
          projectTitle: 'Australian National Interpretive Centre',
          projectDescription: 'Mad Devs engineers helped Sir John Monash Centre to enhance and maintain the BYOD multimedia technology.',
          alt: 'General John Monash The Australian War Memorial`s Logo.'
        },
        {
          projectName: 'guardrails',
          logoImg: 'guardrailsColored',
          link: 'https://blog.maddevs.io/guardrails-helps-800-development-teams-to-establish-devsecops-with-automated-security-reviews-806c3c1d516b?source=friends_link&sk=15f4cdb98f9410b27b9d76dfd94ad1d0',
          targetBlank: true,
          contributionWidgetColors: ['#96969c', '#96969c'],
          projectBackground: 'guardrailsBackground',
          fileExtension: 'png',
          projectColor: '#0e1b27',
          projectTitle: 'Cloud cybersecurity service',
          projectDescription: 'Mad Devs was involved with Guardrails\' security check service as a development contractor with exceptional knowledge of GitHub and GitLab processes.',
          alt: 'Guardrails Cloud CyberSecurity Services Logo.'
        },
        {
          projectName: 'godee',
          logoImg: 'godeeColored',
          link: '/case-studies/godee/',
          targetBlank: false,
          contributionWidgetColors: ['#000', '#000'],
          projectBackground: 'godeeBackground',
          fileExtension: 'png',
          projectColor: '#ff6A01',
          projectTitle: 'Convenient shuttle bus service',
          projectDescription: 'Mad Devs helped GoDee with developing feature-rich software to re-invent public mobility by building new smart ways of a daily commute.',
          alt: 'GoDee Bus Transportation Services Logo.'
        }
      ]
    };
  }
};
</script>

<style lang="scss" scoped>
@import '../../assets/styles/vars';
@import '../../assets/styles/caseIcons';
@import '../../assets/styles/projectsBannerIcons';

.swiper-container {
  background: $bgcolor--project-white;
}

.case-studies {
  &--projects .projects-flame-background {
    position: absolute;
  }

  &__main-title-projects {
    text-align: left !important;

    br {
      display: none;
    }

    @media screen and (max-width: 420px) {
      br {
        display: block;
      }
    }
  }

  .container {
    position: relative;
  }

  &__main-title {
    @include h2_title;
    padding-bottom: 30px;
    color: $text-color--red;
    text-align: center;
    margin: auto;
    display: block;
  }

  .projects {
    display: grid;
    grid-template-columns: repeat(2, 1fr);
    grid-gap: 20px;

    .single-project {
      &:nth-child(2) {
        /deep/ .single-project__background {
          width: auto;
        }
      }

      &:nth-child(3) {
        /deep/ .single-project__background {
          width: auto;
        }

        &::before {
          @include guardrails-background;
          background-repeat: no-repeat;
          background-position: right 50%;
          content: '';
          position: absolute;
          width: 100%;
          height: 430px;
          background-size: contain;
          z-index: -44;
          top: 0;
          right: 0;
          mix-blend-mode: overlay;
          opacity: 0.68;
        }
      }

      &:last-child {
        /deep/ .single-project__background {
          width: auto;
        }

        &::after {
          content: '';
          position: absolute;
          background-image: url('../../assets/img/Studies/jpg/godeeSecondaryBg.jpg');
          background-repeat: no-repeat;
          width: 100%;
          height: 100%;
          background-size: contain;
          z-index: -44;
          top: 0;
          right: 0;
        }
      }
    }

    @media screen and (max-width: 608px) {
      grid-template-columns: repeat(1, 1fr);
    }
  }

  // Projects page, flame icon styles
  &__fire-icon {
    display: block;
    width: 407px;
    height: 661px;
    position: absolute;
    top: -366px;
    right: 100px;
    @include fire-icon;

    @media screen and (max-width: 1320px) {
      width: 340px;
      height: 600px;
      top: -297px;
    }

    @media screen and (max-width: 991px) {
      width: 300px;
      height: 560px;
      top: -253px;
    }

    @media screen and (max-width: 970px) {
      width: 250px;
      height: 510px;
      top: -200px;
    }

    @media screen and (max-width: 834px) {
      top: -255px;
      right: 95px;
    }

    @media screen and (max-width: 768px) {
      top: -255px;
      right: 65px;
    }

    @media screen and (max-width: 690px) {
      width: 240px;
      height: 470px;
      top: -229px;
      right: 55px;
    }

    @media screen and (max-width: 578px) {
      top: -254px;
    }

    @media screen and (max-width: 530px) {
      width: 203px;
      height: 400px;
      top: -197px;
      right: 40px;
    }

    @media screen and (max-width: 480px) {
      width: 145px;
      height: 300px;
      top: -112px;
    }

    @media screen and (max-width: 440px) {
      right: 25px;
    }

    @media screen and (max-width: 420px) {
      top: -70px;
      right: 35px;
    }

    @media screen and (max-width: 355px) {
      right: 30px;
    }
  }
}
</style>
