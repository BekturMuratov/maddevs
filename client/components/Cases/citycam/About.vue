<template>
  <section>
    <section class="container_regular m-48_top media-m-24_top m-48_bottom media-m-24_bottom">
      <TextParagraph>
        Keeping up with the technological evolution, Mad Devs undertook the development of a new-generation
        intersection monitoring system for a company called CityCam. The systemâ€™s purpose is to let anyone
        monitor traffic in a city in real time. Traffic monitoring can help prevent congestion, and overall,
        it makes transportation safer and faster. The project, known internally as Microphotographer, was largely
        about hardware and featured two major parts: engineering the camera device with backend for it and building an
        application with frontend and backend.
      </TextParagraph>
    </section>
    <section class="container_regular m-72_bottom media-m-48_bottom">
      <LottieMad
        id="about-citycam"
        class="case_lottie"
        height="395"
        :lottie-link="$getMediaFromS3(`/images/Cases/clutch/lottie/about-citycam.json`)"
        @animCreated="handleAnimation"
      />
    </section>
  </section>
</template>

<script>
import TextParagraph from '@/components/Cases/shared/TextParagraph'
import LottieMad from '@/components/shared/LottieMad'
import playLottie from '@/helpers/playLottie'

export default {
  name: 'About',
  components: {
    TextParagraph,
    LottieMad,
  },

  methods: {
    handleAnimation(animation) {
      playLottie(animation, 'about-citycam')
    },
  },
}
</script>
