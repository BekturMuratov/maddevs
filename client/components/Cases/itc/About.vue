<template>
  <section class="container_regular m-48_top media-m-24_top">
    <TextParagraph class="m-24_bottom media-m-12_bottom">
      The digital manual is a web-application developed by Mad Devs under the Ready4Trade-Central Asia project, which is a project funded by the European Union and implemented by the International Trade Center (ITC) in the Central Asian region.
    </TextParagraph>
    <TextParagraph>
      The goal of the partnership was to improve the implementation of the Export Management Coaching Initiative (EMCI), a programme where expert coaches in the field of export management train the owners of local small and medium-sized enterprises (SMEs) on how to best optimise business processes and expand trade capacity.
    </TextParagraph>
    <TextQuote class="m-auto m-96_top m-96_bottom media-m-48_top media-m-72_bottom">
      The web application improved reporting and coordination by giving the coaches a convenient tool to collect, process, and report data
    </TextQuote>
  </section>
</template>

<script>
import TextParagraph from '@/components/Cases/shared/TextParagraph'
import TextQuote from '@/components/Cases/shared/TextQuote'

export default {
  name: 'About',
  components: {
    TextParagraph,
    TextQuote,
  },
}
</script>
