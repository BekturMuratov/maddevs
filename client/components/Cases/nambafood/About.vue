<template>
  <section class="container_regular m-48_top m-96_bottom media-m-16_top media-m-48_bottom">
    <TextParagraph>
      Mad Devs developed the Namba Food delivery service from scratch. The apps for couriers, end-users placing orders
      and business owners work as a seamless system, ensuring a smooth delivery process for all types of goods.
    </TextParagraph>
    <TextQuoteBox class="m-48_top m-48_bottom media-m-24_top media-m-24_bottom m-auto">
      Namba Food is a unique delivery service that combines speed, convenience and simplicity.
    </TextQuoteBox>
    <TextParagraph>
      Being first is hard. Namba Foodâ€™s success story focuses on how they took a global idea and successfully localised
      and integrated it into the daily lives of Kyrgyzstan's people. Since 2013, the company has been on a mission to
      bring value to both vendors and customers by conveniently delivering goods at the right time.
    </TextParagraph>
  </section>
</template>

<script>
import TextParagraph from '@/components/Cases/TextParagraph'
import TextQuoteBox from '@/components/Cases/TextQuoteBox'

export default {
  name: 'About',
  components: {
    TextParagraph,
    TextQuoteBox,
  },
}
</script>