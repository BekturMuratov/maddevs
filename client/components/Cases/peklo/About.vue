<template>
  <section class="container_regular m-48_top media-m-24_top">
    <TextParagraph class="m-12_bottom">
      Peklo Tool is a service that simplifies and automates the process of creating contextual advertising campaigns for Google Ads and its Russian-language analogue Yandex Direct. We built it from scratch for Peklo Studio, a digital marketing agency.
    </TextParagraph>
    <TextParagraph class="case_bold-text m-12_bottom">
      Peklo Tool’s two major functions are:
    </TextParagraph>
    <List class="m-72_bottom media-m-48_bottom">
      <ListItemDot>Saving up dozens of working hours in digital marketing specialists’ time</ListItemDot>
      <ListItemDot>Making contextual advertising campaigns measurably more effective</ListItemDot>
    </List>
    <h2 class="case_title_h2 m-12_bottom media-m-24_bottom">
      The needs that Peklo Tool meets
    </h2>
    <TextParagraph class="m-24_bottom media-m-12_bottom">
      Launching contextual ads is a slow, routine, and highly repetitive activity. A marketer or entrepreneur spends days upon days dealing with hundreds of lines in several spreadsheets.
    </TextParagraph>
    <TextParagraph class="m-24_bottom media-m-12_bottom">
      Tedious as it is, the process also features a high risk of errors. When it comes to processing large data arrays, humans are famously prone to mistakes. This work is better left to the reliable machines.
    </TextParagraph>
    <TextParagraph class="m-72_bottom media-m-48_bottom">
      The Peklo Tool team’s effort was to develop an automation solution to speed up the launch of campaigns and rule out human error.
    </TextParagraph>
  </section>
</template>

<script>
import TextParagraph from '@/components/Cases/shared/TextParagraph'
import List from '@/components/Cases/shared/List'
import ListItemDot from '@/components/Cases/shared/ListItemDot'

export default {
  name: 'About',
  components: {
    TextParagraph,
    List,
    ListItemDot,
  },
}
</script>

<style scoped lang="scss">

.case {
  &_bold-text {
    font-weight: 700
  }
}

/deep/ .list-item_dot {
  margin-bottom: 0;
}
</style>
