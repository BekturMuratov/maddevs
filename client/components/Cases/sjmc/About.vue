<template>
  <section
    id="case-first-section"
    class="container_regular m-48_top m-96_bottom media-m-16_top media-m-48_bottom"
  >
    <TextParagraph class="m-24_bottom media-m-12_bottom">
      The Sir John Monash Centre (SJMC) tells its visitors the story about Australia's participation
      on the Western Front. The story is told in the words of those who served.
    </TextParagraph>
    <TextParagraph class="m-24_bottom media-m-12_bottom">
      There is much to learn from the Great War, however, ordinary history books often only explain events on a
      surface level. In contrast, using interactive multimedia installations and technologies, the 480-square metre
      interpretive centre takes visitors on a journey in the footsteps of ordinary soldiers who experienced
      extraordinary events.
    </TextParagraph>
    <TextParagraph>
      The SJMC’s technology enables each visitor to personalise their experience as they learn about the Australian
      soldiers’ stories. Visitors are "fully immersed" in a way that is not seen anywhere else.
    </TextParagraph>
    <TextQuote class="m-auto m-96_top m-96_bottom media-m-48_top media-m-72_bottom">
      World-class BYOD technology delivers an enhanced understanding of the nature and context of the Australian
      contribution to the First World War.
    </TextQuote>
  </section>
</template>

<script>
import TextParagraph from '@/components/Cases/shared/TextParagraph'
import TextQuote from '@/components/Cases/shared/TextQuote'

export default {
  name: 'About',
  components: {
    TextParagraph,
    TextQuote,
  },
}
</script>
