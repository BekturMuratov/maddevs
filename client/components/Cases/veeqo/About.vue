<template>
  <section class="container_regular m-72_top media-m-38_top">
    <TextParagraph>
      Veeqo is an inventory and shipping platform for e-commerce. It helps businesses to manage sales across multiple channels, ship items via multiple carriers, process refunds, manage B2B orders, forecast <span class="case_no-wrap">inventory—the</span> list of features is almost endless. The platform directly integrates with the world’s most popular retail applications, including e-commerce platforms, marketplaces, shipping carriers, POS systems, and more.
    </TextParagraph>
    <TextQuote class="m-auto m-96_top m-96_bottom media-m-48_top media-m-72_bottom">
      Veeqo is a single platform that gives users complete control of their entire inventory. It enables businesses to quickly bulk ship orders from any sales channel, automate repetitive shipping tasks, and track every delivery in one place.
    </TextQuote>
    <TextParagraph>
      Hundreds of retailers all over the world use Veeqo to power their inventory and shipping founded
    </TextParagraph>
  </section>
</template>

<script>
import TextParagraph from '@/components/Cases/shared/TextParagraph'
import TextQuote from '@/components/Cases/shared/TextQuote'

export default {
  name: 'About',
  components: {
    TextParagraph,
    TextQuote,
  },
}
</script>

<style lang="scss" scoped>
  .case {
   &_no-wrap {
      white-space: nowrap;
    }
  }
</style>
