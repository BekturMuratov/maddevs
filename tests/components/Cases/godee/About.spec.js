import About from '@/components/Cases/godee/About.vue'
import { render, screen } from '@testing-library/vue'

describe('About component', () => {
  it('should render correctly', () => {
    const { container } = render(About)

    expect(container).toMatchSnapshot()
  })

  it('should render with text', () => {
    render(About)

    expect(screen.getByText(/The app-based system allows/i).className).toBeTruthy()
  })
})
