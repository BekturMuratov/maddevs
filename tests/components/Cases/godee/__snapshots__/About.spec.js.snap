// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About component should render correctly 1`] = `
<div>
  <section
    class="container_regular"
  >
    <p
      class="case_paragraph m-48_top media-m-24_top"
    >
      
    GoDee is a reliable way to commute in Ho Chi Minh City, Vietnam. Its shuttle buses take express routes that
    link the cityâ€™s major districts and popular destinations. GoDee helps riders save time, money and the planet
    due to its lower ecological impact.
  
    </p>
     
    <div
      class="case_quote-wrap m-auto m-96_top m-96_bottom media-m-48_top media-m-48_bottom"
    >
      <img
        alt="blockquote"
        class="case_mark img_lazy"
        data-src="[object Object]"
        height="50"
        width="50"
      />
       
      <blockquote
        class="case_blockquote"
      >
        
    The app-based system allows users to find a route, select the time, book a seat and pay for the ride online.
  
      </blockquote>
       
      <!---->
    </div>
  </section>
</div>
`;
